app/experimenter/experiments/models.py

```diff
@@ -767,7 +767,7 @@ class NimbusBranch(models.Model):
     class Meta:
         verbose_name = "Nimbus Branch"
         verbose_name_plural = "Nimbus Branches"
-        unique_together = (("slug", "experiment"),)
+        unique_together = (("slug", "experiment", "id"),)
```

Comment:
Oh no this completely breaks the goal of the uniqueness constraint.  The point of the uniqueness constraint is to not have two branches with identical slugs in the same experiment.  Adding `id` to this means you CAN have two branches with the same slug for an experiment, which is the same as not have any uniqueness constraint at all.  And since the branch slug is used for the deterministic enrollment in the client, this would break the enrollment determinism.  Similarly for analysis, branch slugs are used as bases for the analysis so this would also break analysis.

This isn't the right approach.  We have to keep that constraint intact and fix it in the branch saving logic.